src/PrestaShopBundle/Form/Admin/Sell/Discount/DiscountUsabilityType.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShopBundle\Form\Admin\Sell\Discount;
+
+use PrestaShop\PrestaShop\Core\ConstraintValidator\Constraints\UniqueDiscountCode;
+use PrestaShopBundle\Form\Admin\Type\CardType;
+use PrestaShopBundle\Form\Admin\Type\GeneratableTextType;
+use PrestaShopBundle\Form\Admin\Type\TranslatorAwareType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\Constraints\When;
+
+class DiscountUsabilityType extends TranslatorAwareType
+{
+    public const AUTO_MODE = 'auto';
+    public const CODE_MODE = 'code';
+    protected const GENERATED_CODE_LENGTH = 8;
+
+    public function buildForm(FormBuilderInterface $builder, array $options)
```

Comment:
I think this form type will be simplifiable (not sure if the word exists 😅) once my current PR will be merged 

Because the generic component I created used for conditions answers exactly this need:
- one choice selector
- dynamic fields present or not depending on the choice

Reply:
Maybe we should merge your work first, then I do the refacto 🤔
If we can do things better, you might as well do it immediately.